Library/Homebrew/compilers.rb

```diff
@@ -6,6 +6,7 @@ module CompilerConstants
     "gcc-4.0"  => :gcc_4_0,
     "gcc-4.2"  => :gcc,
     "clang"    => :clang,
+    "brew_clang" => :brew_clang,
```

Comment:
Hmmm... That would require quite a few changes to `CompilerFailure`, wouldn't it, since it appears to assume that compiler capabilities can be determined by name and version alone?